### PR TITLE
Automatically publish @celo/contracts and @celo/abis with contract releases

### LINKED ISSUE #10290: Publish a @celo/contracts and @celo/abis
Following discussions with @martinvol and @0xarthurxyz , it would be useful for mento to have the ability to import the core celo contracts directly in solidity.

In order to do so they'd need 
- the `@celo/contracts` to be updated on a regular basis (ideally on push to master)
- a `@celo/abis` to be created and populated with the `build` directory content of the protocol package

Versioning:
- `@celo/contracts` and abis should follow the core contract versioning (eg: they're on v8 right now and working on v9) so the npm package should be 8.x

Automation:
- contracts and abis can be published with the `prerelease` tag on push/merge to master
- for now, because the core contracts release process is still quite manual, the `latest` tag cannot be automated, but should be "easy" to do (`npm dist-tag` could be run by whomever releases core contracts?)

### PR DESCRIPTION
**Pull request by nicolasbrugneaux**
[Closes #10290]

### CODE REVIEW

File: packages/protocol/contracts/README.abis.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+### In your javascript or ts
+
+```js
+import Accounts from '@celo/abis/Accounts.json'
+```
+
+## License
+
+The contents of this package are licensed under the terms of the GNU Lesser Public License V3
```
Comment (aaronmgdr) [inline packages/protocol/contracts/README.abis.md]: can we double check this?

Comment (nicolasbrugneaux) [reply packages/protocol/contracts/README.abis.md]: This is the same license that's in use by the contracts, I suppose the ABIs follow the same license. Happy to verify though.

**Comment by nicolasbrugneaux** [comment]
Blocked until we can get an NPM token.

File: packages/protocol/contracts/README.abis.md (hunk 1 above)
Comment (aaronmgdr) [reply packages/protocol/contracts/README.abis.md]: technically the abis have been included in the contractkit package which has  Apache 2 license.

Comment (aaronmgdr) [reply packages/protocol/contracts/README.abis.md]: I have asked legal

File: .changeset/itchy-tips-repair.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+---
```
Comment (martinvol) [inline .changeset/itchy-tips-repair.md:1]: what is this for?

File: packages/protocol/scripts/bash/prepare-contracts-and-abis-publishing.sh
Comment (martinvol) [inline packages/protocol/scripts/bash/prepare-contracts-and-abis-publishing.sh:1]: Is this meant to be empty?

Comment (nicolasbrugneaux) [reply packages/protocol/scripts/bash/prepare-contracts-and-abis-publishing.sh:1]: wrongly committed file, thanks!

File: .changeset/itchy-tips-repair.md (hunk 2 above)
Comment (nicolasbrugneaux) [reply .changeset/itchy-tips-repair.md:1]: related to changesets #10646 (added by @aaronmgdr ) this is an empty changeset meaning no version change should be required by the packages of the monorepo following this PR

**Comment by nicolasbrugneaux** [comment]
Marking this as blocked by 
- https://github.com/celo-org/celo-monorepo/pull/10647

**Comment by nicolasbrugneaux** [comment]
This is unlocked as of 
- #10690

File: packages/protocol/contracts/package.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "@celo/contracts",
+  "version": "0.0.0-template.version",
+  "author": "cLabs",
+  "license": "LGPL-3.0",
+  "repository": {
+    "type": "git",
+    "url": "hhttps://github.com/celo-org/celo-monorepo/tree/master/packages/protocol/contracts"
```
Comment (aaronmgdr) [inline packages/protocol/contracts/package.json]: typo double "h"

Comment (aaronmgdr) [reply packages/protocol/contracts/package.json]: also i recently learned the correct way is to list the repo with .git and then the directory path 


```
{
  "repository": {
    "type": "git",
    "url": "https://github.com/facebook/react.git",
    "directory": "packages/react-dom"
  }
}
```

File: packages/protocol/tsconfig.json [hunk 4]
```diff
@@ -13,9 +13,10 @@
     "removeComments": false,
     "skipLibCheck": true,
     "sourceMap": true,
-    "target": "es5",
-    "downlevelIteration": true
+    "target": "ES2020",
```
Comment (aaronmgdr) [inline packages/protocol/tsconfig.json]: could this by why things are not working?

File: packages/protocol/scripts/bash/prepare-contracts-and-abis-publishing.sh
Comment (martinvol) [inline packages/protocol/scripts/bash/prepare-contracts-and-abis-publishing.sh:1]: Is this file supposed to be there?

File: packages/protocol/tsconfig.json (hunk 4 above)
Comment (nicolasbrugneaux) [reply packages/protocol/tsconfig.json]: maybe, let me roll it back, pretty sure I changed it for a reason though

File: .github/workflows/celo-monorepo.yml [hunk 5]
```diff
@@ -215,7 +214,7 @@ jobs:
     #   image: us-west1-docker.pkg.dev/devopsre/actions-runner-controller/celo-monorepo:node18
     #   # Required root to install node12
     #   options: --user root
-    timeout-minutes: 30
+    timeout-minutes: 300
```
Comment (soloseng) [inline .github/workflows/celo-monorepo.yml]: This seems like a lot of time to wait

File: .github/workflows/celo-monorepo.yml [hunk 6]
```diff
@@ -302,11 +295,11 @@ jobs:
           sudo apt-get update
           sudo apt-get install vim lsof --yes
           BUILD_AND_DEVCHAIN_DIR=$(echo build/$(echo $RELEASE_TAG | sed -e 's/\//_/g'))
-          yarn --cwd packages/protocol test:devchain-release -b $RELEASE_TAG -d $BUILD_AND_DEVCHAIN_DIR -l /dev/stdout
+          yarn --cwd packages/protocol test:devchain-release -b $RELEASE_TAG -d $BUILD_AND_DEVCHAIN_DIR -l /dev/stdout || echo "failed"
       - name: Setup tmate session
         uses: mxschmitt/action-tmate@v3
         timeout-minutes: 500
```
Comment (soloseng) [inline .github/workflows/celo-monorepo.yml]: ditto

File: .github/workflows/celo-monorepo.yml (hunk 5 above)
Comment (martinvol) [reply .github/workflows/celo-monorepo.yml]: was just for testing, will remove

File: .github/workflows/celo-monorepo.yml (hunk 6 above)
Comment (martinvol) [reply .github/workflows/celo-monorepo.yml]: was just for testing, will remove

**Review by aaronmgdr — APPROVED** [review]
🚀

**Pull request merged**